src/sentry/snuba/metrics/extraction.py

```diff
@@ -1268,7 +1268,10 @@ def to_metric_spec(self, project: Project) -> MetricSpec:
         tag_from_groupbys = self.tags_groupbys(self.groupbys)
         extended_tags_conditions.extend(tag_from_groupbys)
 
-        if self.spec_type == MetricSpecType.DYNAMIC_QUERY:
+        if (
+            self.spec_type == MetricSpecType.DYNAMIC_QUERY
+            and self._tag_for_field("environment") not in extended_tags_conditions
```

Comment:
This also fixes a known issue where the env tag is included more than once.

This does not affect the query hash, thus, it cannot impact any ongoing extraction.